src/calliope/backend/backend_model.py

```diff
@@ -452,39 +451,64 @@ def _add_to_dataset(
                 except KeyError:
                     continue
 
+    @overload
+    def _apply_func(  # noqa: D102, override
+        self,
+        func: Callable,
+        where: Optional[xr.DataArray],
+        n_out: Literal[1],
+        *args,
+        **kwargs,
+    ) -> xr.DataArray: ...
+
+    @overload
+    def _apply_func(  # noqa: D102, override
+        self,
+        func: Callable,
+        where: Optional[xr.DataArray],
+        n_out: Literal[2, 3, 4, 5],
+        *args,
+        **kwargs,
+    ) -> tuple[xr.DataArray, ...]: ...
+
     def _apply_func(
-        self, func: Callable, *args, output_core_dims: tuple = ((),), **kwargs
-    ) -> xr.DataArray:
+        self, func: Callable, where: Optional[xr.DataArray], n_out: int, *args, **kwargs
```

Comment:
Yeah, fair enough, it can always be reintroduced if we want to be able to apply a function without a where but right now all applications have a boolean where array associated with them